lightning-invoice/src/lib.rs

```diff
@@ -904,6 +906,79 @@ impl Deref for PositiveTimestamp {
 }
 
 impl Invoice {
+	/// Utility to construct an invoice.
+	pub fn from_channelmanager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
+		channelmanager: &ChannelManager<Signer, M, T, K, F, L>, amt_msat: Option<u64>, description: String,
+		network: Network,
```

Comment:
Looks like we're taking the dependency on rust-bitcoin specifically to take a network here and then convert it to a Currency, can we just take a Currency?